index.js

```diff
@@ -0,0 +1,60 @@
+document.addEventListener('DOMContentLoaded', () => {
+// console.log("dom")
+setupDroplist();
+fetchMovieList()
+})
+// add list to drop list
+const setupDroplist = ()=>{
+  const getMovieList = document.querySelector('#droplist')
+  getMovieList.addEventListener('click', fetchMovieList)
```

Comment:
You did want to attach an event listener to the select box, however you wanted your listener function to fire on `change` instead of on `click`. A `click` is fired anytime a user clicks the select box even if they don't select a different option. However a `change` event is only fired when the content of the select box changes, that is whenever a user picks a different option. Read about the two events difference's [here](https://www.quora.com/What-is-the-difference-between-onchange-and-onclick-in-JavaScript). You app ends up making a new network request every time a user clicks the select. See your console and network tab in the developer tools to confirm this. How would you change this? Also you don't want to fetch all the movies every time the user selects a different option, right? You only want to fetch all the movies once. Where would be the best place and time to do this? Well, you already have it inside of the listener for `DOMContentLoaded` which is where we want it so that whenever the page loads we populate the select with options. * index.js:13. Judging what this function would do by it's name it seems that `fetchMovieList` and `fetchAllTitles` would be the same/do the same thing. I can see here that you wanted `fetchMovieList` to fetch the movie description and year too. Did you read the API docs? Because there you would see that you won't need any additional network requests to get the movie description and year since all the information comes in the json you get back from `/films/<film-id>`.